integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/utils/PublisherUtils.java

```diff
@@ -163,16 +133,18 @@ public static String createAPI(APIRequest apiRequest, RestAPIPublisherImpl publi
         try {
             createAPIResponse = publisherRestClient.addAPI(apiRequest);
         } catch (ApiException e) {
+            log.error(e.getMessage());
             throw new CCTestException("Error while creating an API", e);
         }
         if (Objects.nonNull(createAPIResponse) && createAPIResponse.getResponseCode() == HttpStatus.SC_CREATED
                 && !StringUtils.isEmpty(createAPIResponse.getData())) {
             log.info("API Created. " + getAPIIdentifierStringFromAPIRequest(apiRequest));
             return createAPIResponse.getData();
         } else {
-            throw new CCTestException(
-                    "Error in API Creation. " + getAPIIdentifierStringFromAPIRequest(apiRequest) + "Response Code:"
-                            + createAPIResponse.getResponseCode() + " Response Data :" + createAPIResponse.getData());
+            String errorMsg = "Error in API Creation. " + getAPIIdentifierStringFromAPIRequest(apiRequest) + "Response Code:"
+                            + createAPIResponse.getResponseCode() + " Response Data :" + createAPIResponse.getData();
+            log.error(errorMsg);
+            throw new CCTestException(errorMsg);
```

Comment:
instead of logging and throwing the same error twice, isn't it better to just throw the error and log the error once it's caught?

Reply:
When the thrown error is logged, it only shows the final cause for the exception unless the entire stack trace is printed. And sometimes during an exception the actual cause can only be found a number of causes behind. Logging the error here to address such a case. But it does look weird. Updating the massage thrown in the exception.

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2624/commits/8f3da3bc19b02e60e2946ddaf269f381f33e527d